libraries/AP_Hygrometer/AP_Hygrometer.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL_Boards.h>
+
+#ifndef HAL_HYGROMETER_ENABLED
+#define HAL_HYGROMETER_ENABLED (BOARD_FLASH_SIZE > 1024)
+#endif
+
+#if HAL_HYGROMETER_ENABLED
+
+#include <stdint.h>
+#include "stdio.h"
+#include <AP_Param/AP_Param.h>
+#include <AP_Logger/AP_Logger.h>
+
+#ifndef HYGROMETER_MAX_SENSORS
+#define HYGROMETER_MAX_SENSORS  2
+#endif
+
+class AP_Hygrometer_Backend;
+
+class AP_Hygrometer
+{
+public:
+    friend class AP_Hygrometer_Backend;
+
+    // constructor
+    AP_Hygrometer();
+
+    void init(void);
+
+    bool get_temperature(uint8_t i, float &temperature) WARN_IF_UNUSED;
+    bool get_temperature(float &temperature) { return get_temperature(primary, temperature); }
+
+    bool get_humidity(uint8_t i, float &humidity);
+    bool get_humidity(float &humidity) { return get_humidity(primary, humidity); }
+
+    bool get_id(uint8_t i, uint8_t &id); 
+    bool get_id(uint8_t &id) { return get_id(primary, id); }
+
+    static const struct AP_Param::GroupInfo var_info[];
+
+    enum class Type {
+        NONE=0,
+        UAVCAN=1,
```

Comment:
```suggestion
#if HAL_ENABLE_LIBUAVCAN_DRIVERS
        UAVCAN=1,
#endif
```

This makes it harder to leave code in accidentally :-)